packages/taco/src/config-validator.ts

```diff
@@ -0,0 +1,279 @@
+      errors.push(
+        `Invalid ritual ID: ${config.ritualId} for domain ${config.domain}`,
+      );
+    }
+
+    // Validate blockchain client configuration
+    if ('viemClient' in config) {
+      // Viem configuration
+      if (!config.viemClient) {
+        errors.push('viemClient is required for viem configuration');
+      }
+      if (!config.viemSignerAccount) {
+        errors.push('viemSignerAccount is required for viem configuration');
+      }
+    } else if ('ethersProvider' in config) {
+      // Ethers configuration
+      if (!config.ethersProvider) {
+        errors.push('ethersProvider is required for ethers configuration');
+      }
+      if (!config.ethersSigner) {
+        errors.push('ethersSigner is required for ethers configuration');
+      }
+    } else {
+      errors.push(
+        'Configuration must include either viem objects (viemClient + viemSignerAccount) or ethers objects (ethersProvider + ethersSigner)',
+      );
+    }
+
+    // Validate chain compatibility (synchronous check)
+    const chainValidation = this.validateChainCompatibility(config);
+    if (!chainValidation.isValid) {
+      errors.push(...chainValidation.errors);
+    }
+
+    return { isValid: errors.length === 0, errors };
+  }
+
+  /**
+   * Synchronous chain compatibility validation
+   *
+   * Validates provider chain compatibility with domain requirements using
+   * synchronously available chain information.
+   *
+   * @param {TacoClientConfig} config - Configuration to validate
+   * @returns {ValidationResult} Validation result
+   * @private
+   */
+  private static validateChainCompatibility(
+    config: TacoClientConfig,
+  ): ValidationResult {
+    const errors: string[] = [];
+
+    // Get expected chain ID for domain
+    const expectedChainId = this.getExpectedChainId(config.domain);
+    if (!expectedChainId) {
+      errors.push(`Unsupported domain: ${config.domain}`);
+      return { isValid: false, errors };
+    }
+
+    // Check viem client chain compatibility
+    if ('viemClient' in config && config.viemClient) {
+      const viemClient = config.viemClient as PublicClient;
+      if (viemClient.chain && viemClient.chain.id !== expectedChainId) {
+        errors.push(
+          `Provider chain mismatch: viem client chain ID ${viemClient.chain.id} does not match domain '${config.domain}' (expected ${expectedChainId})`,
+        );
+      }
+      // Note: If viemClient.chain is undefined, we skip synchronous validation
+      // Full validation with validateFull() will perform the network check
+    }
+
+    // Check ethers provider chain compatibility
+    if ('ethersProvider' in config && config.ethersProvider) {
+      // Note: _network is not public API, but it's the only synchronous way to check
+      // However, if the property `_network` was not available, no error will be thrown.
+      const ethersProvider = config.ethersProvider as unknown as {
```

Comment:
I thought `ethers.Provider` has a `getNetwork()` function? or am I mistaken?

Reply:
Because` _network` is private there is no relying on it, the code will use it, if and only if, it was available. And if it was not available or if reading it throws an exception, the exception is caught, so no error will be thrown, and no error will be reported in the validation results as well.

Yes, the `getNetwork()` is called inside `validate()`. But for `validateFast()` that is called as early as in the constructor, checking the response of an async is not possible.